### PR TITLE
Add custom vpc support in AWS cloud prepare

### PR DESCRIPTION
**Pull request by aswinsuryan**
Add custom vpc support in AWS cloud prepare

### CODE REVIEW

File: out.txt [hunk 1]
```diff
@@ -0,0 +1,443 @@
+/*
```
Comment (tpantelis) [inline out.txt]: I assume you didn't mean to submit this file.

File: pkg/aws/aws.go [hunk 2]
```diff
@@ -40,36 +41,84 @@ const (
...
+	}
+}
+
 type awsCloud struct {
 	client               awsClient.Interface
 	infraID              string
 	region               string
 	nodeSGSuffix         string
 	controlPlaneSGSuffix string
+	CloudConfig          map[string]interface{}
```
Comment (tpantelis) [inline pkg/aws/aws.go]: This doesn't need to be public for consistency.

```suggestion
	cloudConfig          map[string]interface{}
```

File: pkg/aws/ocpgwdeployer.go [hunk 3]
```diff
@@ -69,16 +69,35 @@ func (d *ocpGatewayDeployer) Deploy(input api.GatewayDeployInput, status reporte
...
+				publicSubnets = append(publicSubnets, *subnet)
+			}
+		} else {
+			return errors.New("Subnet IDs must be a valid non-empty slice of strings")
+		}
+	} else {
+		publicSubnets, err = d.aws.findPublicSubnets(vpcID, d.aws.filterByName("{infraID}*-public-{region}*"))
+		if err != nil {
+			return status.Error(err, "unable to find public subnets")
+		}
```
Comment (tpantelis) [inline pkg/aws/ocpgwdeployer.go:100]: This code block looks the same as the block starting on L152 above except the latter checks for `len(publicSubnets)`. I think we can refactor to a function and let the caller check `len(publicSubnets)`.

File: pkg/aws/ocpgwdeployer.go [hunk 4]
```diff
@@ -97,9 +116,20 @@ func (d *ocpGatewayDeployer) Deploy(input api.GatewayDeployInput, status reporte
 
 	status.Success("Created Submariner gateway security group %s", gatewaySG)
 
+	err = d.processSubnets(vpcID, gatewaySG, publicSubnets, input, status)
+	if err != nil {
+		return err
+	}
+
+	return nil
```
Comment (tpantelis) [inline pkg/aws/ocpgwdeployer.go]:
```suggestion
	return d.processSubnets(vpcID, gatewaySG, publicSubnets, input, status)
```

**Review by vthapar — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**